### ISSUE TITLE
Are the H-only checkpoints available?

### ISSUE THREAD

**Issue by newalexander** [MarklandGroup/NMR2Struct#10]
Thanks for making this work available. In the paper, you talk about models trained on only 1H spectra, only 13C spectra, and both (Table 1). Are all of those checkpoints on this repo, or is there only the best-performing configuration? It would be great to have the checkpoint for the 1H-only model if it's compatible with the public codebase.

**Comment by FranklinHu1** [comment]
H- and C-NMR only checkpoints are now uploaded to the checkpoints folder. Closing for now, but feel free to re-open if you encounter any issues!

**Comment by newalexander** [comment]
Great, thank you! We'll start working with this trained model.

We also tried retraining the model using only H-NMR data on our system. With the below training and inference configs, and using your paper's datasets and splits, we got down to a minimum validation loss of about 0.3 after ~400 epochs. However, this model only got a top-15 structure prediction accuracy of ~38% on the test set, compared to the expected ~60%.

If you have a chance to look at the configs, can you see if we've set up anything incorrectly, or if there are any other obvious issues? We're hoping to explore fine-tuning on some of our own data in the future, so it would be great to make sure we understand the code base as best as we can.

Training:
```yaml
global_args:
  ngpus: 1
  dtype: float32
  savedir: "..."
  seed: 42

data:
  spectra_file: 
    - '.../spectrum_to_struct_dset/spectra.h5'
  label_file: 
    - '.../spectrum_to_struct_dset/substructures.h5'
  smiles_file: 
    - '.../spectrum_to_struct_dset/smiles.npy'
  input_generator: "SpectrumRepresentationUnprocessed"
  input_generator_addn_args: {}
  target_generator: "SMILESRepresentationTokenizedWithSubstructs"
  target_generator_addn_args: {}
  alphabet: '.../NMR2Struct/example_configs/alphabet.npy'
  front_load_data_processing: True
  eps: 0.005

model:
  model_type: "MultiTaskModel"
  load_model: null
  model_args:
    src_embed: ConvolutionalEmbedding
    src_embed_options:
      d_model: 128
      n_hnmr_features: 28000
      n_cnmr_features: 80
      use_hnmr: True
      use_cnmr: False
    forward_fxn: src_fwd_fxn_conv_embedding
    substructure_model: "EncoderModel"
    substructure_model_args:
      d_model: 128
      d_out: 957
      dim_feedforward: 1024
      nhead: 4
      num_layers: 4
      output_head: SingleLinear
      output_head_opts:
        d_model: 128
        d_out: 957
      pooler: SeqPool
      pooler_opts:
        d_model: 128
      source_size: 958
      src_embed: null
      src_embed_options: {}
      src_forward_function: src_fwd_fxn_packed_tensor
      src_pad_token: null
    structure_model: "TransformerModel"
    structure_model_args: 
      src_embed: null
      src_embed_options: {}
      tgt_embed: "nn.embed"
      tgt_embed_options: {}
      src_forward_function: "src_fwd_fxn_packed_tensor"
      tgt_forward_function: "tgt_fwd_fxn_basic"
      source_size: 28045
      target_size: 24
      d_model: 128
      dim_feedforward: 1024
      src_pad_token: null
      tgt_pad_token: 21
    substructure_model_ckpt: null
    structure_model_ckpt: '.../NMR2Struct/checkpoints/multitask_checkpoint.pt'

training:
  nepochs: 1500
  top_checkpoints_n: 10
  checkpoint_loss_metric: 'val_loss_0'
  write_freq: 100
  test_freq: 10
  prev_epochs: 0
  splits: 
    - '.../spectrum_to_struct_dset/split_indices.p'
  train_size: 0.8
  val_size: 0.1
  test_size: 0.1
  optimizer: 'Adam'
  optimizer_args:
    lr: 0.00001
  scheduler: null
  dloader_args:
    batch_size: 64
    shuffle: true
  loss_fn: "MultiTaskLoss"
  loss_fn_args: 
    ignore_index: null
    substructure_weight: 1
    structure_weight: 1
```


Inference:
```yaml
global_args:
  ngpus: 1
  dtype: float32
  savedir: "..."
  seed: 42

data:
  spectra_file: 
    - '.../spectrum_to_struct_dset/spectra.h5'
  label_file: 
    - '.../spectrum_to_struct_dset/substructures.h5'
  smiles_file: 
    - '.../spectrum_to_struct_dset/smiles.npy'
  input_generator: "SpectrumRepresentationUnprocessed"
  input_generator_addn_args: {}
  target_generator: "SMILESRepresentationTokenizedWithSubstructs"
  target_generator_addn_args: {}
  alphabet: null #Populate
  front_load_data_processing: True
  eps: 0.005

model:
  model_type: "MultiTaskModel"
  load_model: null
  model_args:
    src_embed: ConvolutionalEmbedding
    src_embed_options:
      d_model: 128
      n_hnmr_features: 28000
      n_cnmr_features: 80
      use_hnmr: True
      use_cnmr: False
    forward_fxn: src_fwd_fxn_conv_embedding
    substructure_model: "EncoderModel"
    substructure_model_args:
      d_model: 128
      d_out: 957
      dim_feedforward: 1024
      nhead: 4
      num_layers: 4
      output_head: SingleLinear
      output_head_opts:
        d_model: 128
        d_out: 957
      pooler: SeqPool
      pooler_opts:
        d_model: 128
      source_size: 958
      src_embed: null
      src_embed_options: {}
      src_forward_function: src_fwd_fxn_packed_tensor
      src_pad_token: null
    structure_model: "TransformerModel"
    structure_model_args: 
      src_embed: null
      src_embed_options: {}
      tgt_embed: "nn.embed"
      tgt_embed_options: {}
      src_forward_function: "src_fwd_fxn_packed_tensor"
      tgt_forward_function: "tgt_fwd_fxn_basic"
      source_size: 28045
      target_size: 24
      d_model: 128
      dim_feedforward: 1024
      src_pad_token: null
      tgt_pad_token: 21
    substructure_model_ckpt: null
    structure_model_ckpt: null


inference:
  model_selection: '.../model_epoch=425_loss=0.29909850.pt'
  splits: 
    - '.../spectrum_to_struct_dset/split_indices.p'
  train_size: 0.8 
  val_size: 0.1 
  test_size: 0.1 
  dloader_args:
    shuffle: false
    batch_size: 1024
  sets_to_run: ['test']
  run_inference_args: 
    pred_gen_fn: 'infer_transformer_model'
    pred_gen_opts:
      num_pred_per_tgt: 15
      sample_val: 5
      tgt_start_token: 22
      tgt_stop_token: 23
      track_gradients: true
      alphabet: '.../example_configs/alphabet.npy'
      decode: true
      infer_fwd_fxn: multitask
  write_frequency: 100
```

**Comment by FranklinHu1** [comment]
I've reopened the issue. Based on your config, one of the first things that jumps out at me is the value for the field `model.model_args.structure_model_ckpt` in your training configuration. This should not be set to the multitask model checkpoint, but rather to the `transformer_checkpoint.ckpt` file provided in the repository. The `multitask_checkpoint.pt` model is a checkpoint for the overall model trained on both H and C NMR spectra. If you want to reproduce the number in the paper, you will need to use the `transformer_checkpoint.ckpt` instead.

Give that a shot and feel free to follow up with any other issues/concerns. Excited to see how the fine-tuning turns out!

**Comment by newalexander** [comment]
Thanks, appreciate the clarification, that would definitely be important. We'll retrain with the right checkpoint and check results. Then the H-NMR only checkpoint does let us replicate paper results, so thanks for that!
